src/java.desktop/share/classes/javax/swing/text/DefaultStyledDocument.java

```diff
@@ -497,7 +497,7 @@ public Style getLogicalStyle(int p) {
      *  before setting the new attributes
      */
     public void setCharacterAttributes(int offset, int length, AttributeSet s, boolean replace) {
-        if (length == 0) {
+        if (length <= 0) {
```

Comment:
@param length the length &gt;= 0 javadoc should it not be then > 0 ?

Reply:
Yeah, its should be > 0 in java doc. Will update it accordingly. Then should I raise a CSR......?

Reply:
probably yes

Reply:
Updated.